analysis/trigger_performance/PHL_trigger_report.Rmd

```diff
@@ -0,0 +1,349 @@
+trigger need to be reached."
+```
+
+```{r setup, include = FALSE, message = FALSE, warning = FALSE}
+knitr::opts_chunk$set(echo = FALSE) # do not print code by default
+knitr::opts_chunk$set(include = TRUE) # include chunk output by default
+knitr::opts_chunk$set(message = FALSE) # do not print messages by default
+knitr::opts_chunk$set(warning = FALSE) # do not print warnings by default
+
+options(scipen = 999) # turn off scientific notation
+options(encoding = "UTF-8") # set encoding to UTF-8 instead of ANSI
+```
+
+```{r libraries, include = FALSE, warning = FALSE}
+packages <- c('tidyverse', 'ggthemes', 'kableExtra', 'knitr', 'flextable')
+
+# install packages not yet installed
+installed_packages <- packages %in% rownames(installed.packages())
+
+if (any(installed_packages == FALSE)) {
+  install.packages(packages[!installed_packages])
+}
+
+# load libraries
+lapply(packages, library, character.only = TRUE)
+
+```
+
+
+```{r paths}
+data_folder_path <- paste0(Sys.getenv("AA_DATA_DIR"), "/public/exploration/", country_pcode3,"/trigger_performance/")
+perf_metrics_95_filename <- paste0(data_folder_path, country_pcode3, "_perf_metrics_table_ci_0.95.csv")
+perf_metrics_68_filename <- paste0(data_folder_path, country_pcode3, "_perf_metrics_table_ci_0.68.csv")
+learning_table_filename <- paste0(data_folder_path, country_pcode3, "_learning_table.csv")
+explanatory_ci_plot_path <- ("country_template/docs/explanatory_ci.png")
+```
+
+```{r functions}
+source("country_template/docs/plotAtvCI.R")
+source("country_template/docs/plotTradeoffCI.R")
+```
+
+```{r data-loading-and-formatting}
+
+# read in performance metrics files
+perf_metrics_95_data <- read.csv(perf_metrics_95_filename,
+                              colClasses = c('character', 'character', 'numeric', 'character')
+                              )
+
+perf_metrics_68_data <- read.csv(perf_metrics_68_filename,
+                              colClasses = c('character', 'character', 'numeric', 'character')
+                              )
+
+# format performance metrics values
+perf_metrics_95_data$value <- round(100 * perf_metrics_95_data$value, 0)
+perf_metrics_68_data$value <- round(100 * perf_metrics_68_data$value, 0)
+
+# uniquely identify point labels
+perf_metrics_95_data$upoint <- paste0(perf_metrics_95_data$point, "_95")
+perf_metrics_68_data$upoint <- paste0(perf_metrics_68_data$point, "_68")
+
+# combine datasets
+perf_metrics_data <- rbind(perf_metrics_95_data, perf_metrics_68_data) %>%
+                      select(-point) # remove non-unique point identifier
+
+
+```
+
+
+```{r atv-CIs}
+
+# Create a CI visual for atv for each trigger, activation timepoint, and activation scenario beside None and Any
+
+trigger_list = c("readiness", "activation", "framework")
+
+for(trigger in trigger_list) {
+
+  # produce segment_dimensions and plots for triggers
+  output <- plotAtvCI(trigger_id = trigger)
+
+  # save plot as png
+  filename <- paste0(trigger, "_atv.png")
```

Comment:
I believe I figured out a way to write and read from `Data` instead of pushing plots to GH. I'll test and update the template file accordingly, and we can then update this code for the PHL.